webui/src/lib/api/index.js

```diff
@@ -996,11 +999,11 @@ class BranchProtectionRules {
 
     }
 
-    async createRule(repoID, pattern) {
-        const response = await apiRequest(`/repositories/${encodeURIComponent(repoID)}/branch_protection`, {
-            method: 'POST',
-            body: JSON.stringify({pattern: pattern})
-        });
+    async setRules(repoID, rules, lastKnownChecksum) {
+        const response = await apiRequest(`/repositories/${encodeURIComponent(repoID)}/settings/branch_protection`, {
+            method: 'PUT',
+            body: JSON.stringify(rules),
+        }, {'ETag': lastKnownChecksum});
```

Comment:
Are we sure we want the name to be ETag, it's basically an implementation detail, for the user it's just a lock/token/predicate or something like that

Reply:
I do not care much either way: ETag is by now a term of HTTP, not only of S3.  It's an actual [standard header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag).

Reply:
UPDATE: this was a mistake, the header used here should be `If-Match`.